.circleci/config.yml

```diff
@@ -175,11 +211,20 @@ workflows:
   version: 2
   workflow:
     jobs:
+#       This job will run on every commit on any branch
       - build_debug
-      - test_unit
-      - test_instrumented:
+#       Unit tests should only start after a successful build. Or vice versa?
+      - test_unit:
+#         Although the unit testing task could run in parallel with build task,
+#         in such case, it would not benefit from the previous caching in the event of
+#         any change in the .gradle script files.
+#         Hence,
           requires:
             - build_debug
+      # Instrumentation tests should only start after unit tests complete gracefully. Or vice versa?
```

Comment:
What are the trade offs. If it's something to try empirically, then might be easier to file an issue for a contributor to try.

Reply:
It's mostly an organizational policy than a technical decision. There's no direct dependency between these two jobs. However, it's likely that in case one fails, the other would fail too. So one matter here is wasting (cloud) resources when not needed and on the other side the counterweight is if we care about possibly wasting these resources on failed tests more than we care about finishing the builds quicker in the event of a success?

Reply:
Finishing faster at all costs. We currently don't have to worry about waste.